packages/provider/lib/src/async_provider.dart

```diff
@@ -164,15 +164,15 @@ Exception:
 ${snapshot.error}
 ''');
   }
-  return snapshot.data;
+  return snapshot.hasData ? snapshot.data : null;
```

Comment:
There's no reason to check `hasData` here. This is redundant with `initialData`

Reply:
At tip-of-tree Flutter, the deprecated `initialData` field will trigger a "no data" snapshot if the `initialData` is  null, so the check is still needed.

Once the Flutter change has made it through to the stable channel, you can update this use `.withInitialData()`, and then the `hasData` check won't be needed anymore.

Reply:
(for this reason, without the changes in this PR, the tests fail when run against Flutter with https://github.com/flutter/flutter/pull/36577 applied)

Reply:
But on master, the. `provider` would use `StreamBuilder.withInitialData`.

I'd prefer to stop using `StreamBuilder` entirely if that's an issue. `provider` voluntarily don't expose `AsyncSnashot` to begin with.

Reply:
I see.  So the other way to go would be https://github.com/rrousselGit/provider/pull/166, which puts an upper bound on the Flutter SDK to prepare for  the breaking change.  Then this would be updated to use the new API instead  of the deprecated one (and would be updated to version `3.1.2` and would contain a _lower bound_ Flutter SDK constraint).

wdyt?

Reply:
Hum I don't like that either.
Using StreamBuilder here is just an implementation details. 

The best alternative is for `provider` to stop using `StreamBuilder`, and do the listening itself.
That should even give a small performance boost, considering `provider` don't need `AsyncSnashot` at all.

Reply:
Ok.  I'm not familiar with the implementation of `provider` -- are you game to update it to not use `StreamBuilder`?

Reply:
Sure. I'll do that.. Although depending on where https://github.com/flutter/flutter/issues/34545#issuecomment-513482773 is going, there may be no need for a change at all.